judgels-frontends/raphael/src/components/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+
+import { stringify, parse } from 'query-string';
+import { push } from 'react-router-redux';
+import { withRouter, RouteComponentProps } from 'react-router';
+import { connect } from 'react-redux';
+import SearchBoxForm, { SearchBoxFormData } from './SearchBoxForm';
+
+interface SearchBoxRouteProps extends RouteComponentProps<{}> {
+  onAppendRoute: (queries: any) => any;
+}
+
+export interface SearchBoxProps {
+  onSubmit?: (content: string) => any;
+  nextRoute?: (content: string, prevQueries: any) => any;
```

Comment:
Why is this optional?

Reply:
Actually I intended this component to be flexible and can be used in the other places, such as user table.

Making it optional allows it to be more flexible (suppose we don't want to put the filter param in path)